metricflow/data_table/mf_table.py

```diff
@@ -0,0 +1,279 @@
+                        "cell_value": cell_value,
+                    },
+                )
+                # Check that datetimes don't have a timezone set.
+                if isinstance(cell_value, datetime.datetime):
+                    assert cell_value.tzinfo is None, mf_pformat_many(
+                        "Time zone provided for datetime.",
+                        {
+                            "row_index": row_index,
+                            "column_index": column_index,
+                            "cell_value": cell_value,
+                        },
+                    )
+
+    @property
+    def column_count(self) -> int:  # noqa: D102
+        return len(self.column_descriptions)
+
+    @property
+    def row_count(self) -> int:  # noqa: D102
+        return len(self.rows)
+
+    def column_name_index(self, column_name: str) -> int:
+        """Return the index of the column that matches the given name. Raises `ValueError` if the name is invalid."""
+        for i, column_description in enumerate(self.column_descriptions):
+            if column_description.column_name == column_name:
+                return i
+        raise ValueError(
+            f"Unknown column name {repr(column_name)}. Known column names are:"
+            f"\n{indent(mf_pformat([column_name for column_name in self.column_descriptions]))}"
+        )
+
+    @property
+    def column_names(self) -> Sequence[str]:  # noqa: D102
+        return tuple(column_description.column_name for column_description in self.column_descriptions)
+
+    def column_values_iterator(self, column_index: int) -> Iterator[CellValue]:
+        """Returns an iterator for values of the column at the tiven index."""
+        return (row[column_index] for row in self.rows)
+
+    def _sorted_by_column_name(self) -> MetricFlowDataTable:  # noqa: D102
+        # row_dict_by_row_index: Dict[int, Dict[str, CellType]] = defaultdict(dict)
+
+        new_rows: List[List[CellValue]] = [[] for _ in range(self.row_count)]
+        sorted_column_names = sorted(self.column_names)
+        for column_name in sorted_column_names:
+            old_column_index = self.column_name_index(column_name)
+            for row_index, cell_value in enumerate(self.column_values_iterator(old_column_index)):
+                new_rows[row_index].append(cell_value)
+
+        return MetricFlowDataTable(
+            column_descriptions=tuple(
+                self.column_descriptions[self.column_name_index(column_name)] for column_name in sorted_column_names
+            ),
+            rows=tuple(
+                tuple(row_dict[column_index] for column_index in range(self.column_count)) for row_dict in new_rows
+            ),
+        )
+
+    def _sorted_by_row(self) -> MetricFlowDataTable:  # noqa: D102
+        def _cell_sort_key(cell: CellValue) -> str:
+            if isinstance(cell, datetime.datetime):
+                return cell.isoformat()
+            return str(cell)
```

Comment:
Is this potentially non-deterministic for floats?

Reply:
This could be tricky - did you have thoughts on an approach?